sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -97,18 +98,34 @@ object GpuParquetFileFormat {
     }
   }
 
-  def getPrecisionList(schema: StructType): Seq[Int] =  {
-    def precisionsList(t: DataType): Seq[Int] = {
-      t match {
-        case d: DecimalType => List(d.precision)
-        case _: StructType =>
-          throw new IllegalStateException("structs are not supported right now")
-        case _: ArrayType =>
-          throw new IllegalStateException("arrays are not supported right now")
-        case _ => List(0)
+  def parquetWriterOptionsFromSchema[T <: NestedBuilder[_, _], V <: ParquetColumnWriterOptions]
+  (builder: ParquetColumnWriterOptions.NestedBuilder[T, V], schema: StructType): T = {
+    // TODO once https://github.com/rapidsai/cudf/issues/7654 is fixed go back to actually
+    // setting if the output is nullable or not everywhere we have hard-coded nullable=true
+    schema.foreach(field =>
+      field.dataType match {
+        case dt: DecimalType =>
+          builder.withDecimalColumn(field.name, dt.precision, true)
+        case TimestampType =>
+          builder.withTimestampColumn(field.name,
+            // TODO once https://github.com/rapidsai/cudf/issues/8070 is fixed uncomment the line
```

Comment:
We need a P1 blocker issue filed against us for this too.  At a minimum we need to fall back to the CPU for this.

Reply:
I created this when I saw the issue. https://github.com/rapidsai/cudf/issues/8070

Reply:
This is the plugin bug tracking this https://github.com/NVIDIA/spark-rapids/issues/2302